### PR TITLE
Enable more code linters

### PR DESCRIPTION
**Pull request by sergefdrv**

### CODE REVIEW

**Review by matejpavlovic — APPROVED** [review]

**Review by dnkolegov — COMMENTED** [review]
Why not https://staticcheck.io/docs/checks/ ?

**Comment by sergefdrv** [comment]
> Why not https://staticcheck.io/docs/checks/ ?

It is [enabled by default](https://golangci-lint.run/usage/linters/#enabled-by-default).

**Comment by sergefdrv** [comment]
The effective set of enabled linters in this cofiguration:
```
$ golangci-lint linters
Enabled by your configuration linters:
deadcode: Finds unused code [fast: false, auto-fix: false]
dupl: Tool for code clone detection [fast: true, auto-fix: false]
errcheck: Errcheck is a program for checking for unchecked errors in go programs. These unchecked errors can be critical bugs in some cases [fast: false, auto-fix: false]
goconst: Finds repeated strings that could be replaced by a constant [fast: true, auto-fix: false]
gocyclo: Computes and checks the cyclomatic complexity of functions [fast: true, auto-fix: false]
gofmt: Gofmt checks whether code was gofmt-ed. By default this tool runs with -s option to check for code simplification [fast: true, auto-fix: true]
gosec (gas): Inspects source code for security problems [fast: false, auto-fix: false]
gosimple (megacheck): Linter for Go source code that specializes in simplifying a code [fast: false, auto-fix: false]
govet (vet, vetshadow): Vet examines Go source code and reports suspicious constructs, such as Printf calls whose arguments do not align with the format string [fast: false, auto-fix: false]
ineffassign: Detects when assignments to existing variables are not used [fast: true, auto-fix: false]
misspell: Finds commonly misspelled English words in comments [fast: true, auto-fix: true]
revive: Fast, configurable, extensible, flexible, and beautiful linter for Go. Drop-in replacement of golint. [fast: false, auto-fix: false]
staticcheck (megacheck): Staticcheck is a go vet on steroids, applying a ton of static analysis checks [fast: false, auto-fix: false]
structcheck: Finds unused struct fields [fast: false, auto-fix: false]
typecheck: Like the front-end of a Go compiler, parses and type-checks Go code [fast: false, auto-fix: false]
unconvert: Remove unnecessary type conversions [fast: false, auto-fix: false]
unused (megacheck): Checks Go code for unused constants, variables, functions and types [fast: false, auto-fix: false]
varcheck: Finds unused global variables and constants [fast: false, auto-fix: false]
```

**Review by dnkolegov — APPROVED** [review]

**Review by matejpavlovic — APPROVED** [review]

File: cmd/mircat/debug.go [hunk 1]
```diff
@@ -128,9 +129,9 @@ func debuggerNode(id t.NodeID, membership []t.NodeID) (*mir.Node, error) {
 	})
 	if err != nil {
 		return nil, fmt.Errorf("could not instantiate mir node: %w", err)
-	} else {
```
[the remark this answers is not in the document]
Comment (sergefdrv) [reply cmd/mircat/debug.go:131]: I think the idea behind this warning is that it's often more readable when the normal flow is less indented than the error handling

[the remark this answers is not in the document]
Comment (matejpavlovic) [reply cmd/mircat/debug.go:131]: Fair enough, as I said, it's purely a matter of personal taste. No strong feelings about it, absolutely fine with removing the `else` branch.

**Pull request merged**